adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -617,10 +647,28 @@ func createUpstreamTLSContext(upstreamCerts []byte, address *corev3.Address) *tl
 			}
 		}
 
-		upstreamTLSContext.CommonTlsContext.ValidationContextType = &tlsv3.CommonTlsContext_ValidationContext{
-			ValidationContext: &tlsv3.CertificateValidationContext{
-				TrustedCa: trustedCASrc,
-			},
+		// Sni should be assigned when there is a hostname
+		if isRateLimitCluster {
+			upstreamTLSContext.Sni = conf.Envoy.RateLimit.SSLCertSANHostname
```

Comment:
Please refer to https://github.com/wso2/product-microgateway/pull/3142/files#r1044083874. Lets replace the entire upstreamTLSContext if required.